### PR TITLE
Use a realistic dataset

### LINKED ISSUE #2225: Use a real chart of accounts in test data
I've run into a couple errors based on accounts not having the proper type.  For example, the cashbox account is currently a title account.  This makes testing difficult - tests fail not because the logic is incorrect, but because the data is improperly made.

We could improve our tests greatly by using an actual chart of accounts, such as the one in the development dataset.  This would allow us to reason about our tests in a much more profound way and avoid errors based on improperly created data.

[Motivated by this annoyance](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2226#discussion_r145977608).

### PR DESCRIPTION
**Pull request by jniles**
This PR upgrades the test dataset of the application to be more realistic.  This includes:
 1. An OHADA compliant chart of accounts (as far as I know)
 2. A complete set of inventory items and groups

The remainder of the dataset is updated to point at the correct accounts and inventory items.  Integration tests have been updated and end to end tests are currently a work in progress.  Where possible, I have also updated the test syntax to pass ESLint checks.

Closes #2225.

### CODE REVIEW

**Comment by jniles** [comment]
@mbayopanda, this is now ready for review.

**Comment by jniles** [comment]
@mbayopanda, can I get a review?

File: client/src/modules/accounts/templates/grid.actionsCell.tmpl.html [hunk 1]
```diff
@@ -3,6 +3,6 @@
     id="account-edit-{{row.entity.id}}"
     ui-sref="accounts.edit({id:row.entity.id})">
     <i class="fa fa-edit"></i>
-    {{ "TABLE.COLUMNS.EDIT" | translate }} {{row.entity.number}}
+    <span translate>TABLE.COLUMNS.EDIT</span> {{row.entity.number}}
```
Comment (mbayopanda) [inline client/src/modules/accounts/templates/grid.actionsCell.tmpl.html:6]: 👍

File: client/src/modules/cash/cashboxes/configure_currency/modal.html [hunk 2]
```diff
@@ -26,6 +26,7 @@
       account-id="CashboxModalCtrl.data.account_id"
       label="FORM.LABELS.CASH_ACCOUNT"
       name="account_id"
+      exclude-title-accounts="true"
```
Comment (mbayopanda) [inline client/src/modules/cash/cashboxes/configure_currency/modal.html:29]: Very nice 👍

File: client/src/modules/inventory/configuration/groups/groups.js [hunk 3]
```diff
@@ -45,11 +45,15 @@ function InventoryGroupsController($translate, InventoryGroup, Account, Notify,
...
-    .catch(Notify.handleError);
+      .then(function (res) {
+        Notify.success('FORM.INFO.UPDATE_SUCCESS');
+      })
+      .then(startup)
+      .catch(function (err) {
+        if (err) {
+          Notify.handleError(err);
+        }
+      });
```
Comment (mbayopanda) [inline client/src/modules/inventory/configuration/groups/groups.js:56]: What is wrong with ?
```js
.catch(Notify.handleError);
```

File: client/src/modules/inventory/configuration/groups/modals/actions.modal.js [hunk 4]
```diff
@@ -79,53 +81,13 @@ function InventoryGroupsActionsModalController(Account, InventoryGroup, Notify,
...
-          .then(function (account) {
-            vm.session.cogsAccount = essentialAccountDetail(account);
-          })
-          .catch(Notify.handleError);
-        }
-      })
-      .catch(Notify.handleError);
+      InventoryGroups.read(vm.identifier)
+        .then(function (groups) {
+          console.log(groups);
```
Comment (mbayopanda) [inline client/src/modules/inventory/configuration/groups/modals/actions.modal.js:87]: Is it possible to remove this `console.log(groups)` ?

File: client/src/modules/inventory/configuration/groups/modals/actions.tmpl.html [hunk 5]
```diff
@@ -37,78 +37,39 @@
...
-      </ui-select>
+    <bh-account-select
+      name="sales_account"
+      id="sales_account"
+      account-id="$ctrl.session.sales_account"
+      exclude-title-accounts="true"
+      on-select-callback="$ctrl.onSelectSalesAccount(account)">
+      (<span translate>FORM.LABELS.SALE</span>)
+      <bh-clear on-clear="$ctrl.session.sales_account = null;"></bh-clear>
+    </bh-account-select>
```
Comment (mbayopanda) [inline client/src/modules/inventory/configuration/groups/modals/actions.tmpl.html:48]: 💃

File: test/end-to-end/cashboxes/cashboxes.spec.js [hunk 6]
```diff
@@ -17,11 +17,8 @@ describe('Cashboxes', () => {
     project : 'Test Project A',
   };
 
-  // clicks the 'update' button on the cashbox at $index n in the table
-  function update(n) {
-    return element(by.repeater('box in CashCtrl.cashboxes track by box.id').row(n))
-      .$$('a')
-      .click();
+  function update(label) {
+    $(`[data-cashbox="${label}"]`).$$('a').click();
```
Comment (mbayopanda) [inline test/end-to-end/cashboxes/cashboxes.spec.js:21]: 👍

File: test/integration/inventory.js [hunk 7]
```diff
@@ -216,9 +213,9 @@ describe('(/inventory) The Inventory HTTP API', () => {
   });
 
   // delete the inventory unit
-  it('DELETE /inventroy/units delete an existing inventory unit', () => {
```
Comment (mbayopanda) [inline test/integration/inventory.js:219]: 😆

**Review by mbayopanda — APPROVED** [review]
LGTM

File: test/data.sql [hunk 8]
```diff
@@ -1,33 +1,251 @@
...
+  (266, 6, 1, 776, 'GAINS DE CHANGE', 62, 0, NULL, NULL, '2016-10-23 16:05:34', NULL, 0),
+  (267, 4, 1, 77611010, 'Gain de change *', 266, 0, NULL, 1, '2016-10-23 16:05:34', NULL, 0),
+  (280, 5, 1, 81111010, 'Compte Immobilisations incorporelles', 65, 0, NULL, NULL, '2016-10-23 16:05:34', NULL, 0),
+  (281, 5, 1, 81211010, 'Compte Immobilisations corporelles', 65, 0, NULL, NULL, '2016-10-23 16:05:34', NULL, 0),
+  (282, 5, 1, 81611010, 'Compte Immobilisations financières', 65, 0, NULL, NULL, '2016-10-23 16:05:34', NULL, 0),
+  (283, 3, 1, 13110001, 'Résusltat de l\'exercise', 111, 0, NULL, NULL, '2017-06-09 12:29:04', NULL, 0),
+  (284, 1, 1, 40111000, 'SNEL SUPPLIER', 170, 0, NULL, NULL, '2016-10-23 16:05:34', NULL, 0),
+  (285, 1, 1, 40111001, 'REGIDESO SUPPLIER', 170, 0, NULL, NULL, '2016-10-23 16:05:34', NULL, 0),
+  (300, 5, 1, 40111002, 'SUPPLIER\'S ACCOUNT 1', 170, 0, NULL, NULL, '2017-11-06 15:07:21', NULL, 0),
+  (301, 5, 1, 40111003, 'SUPPLIER\'S ACCOUNT 2', 170, 0, NULL, NULL, '2017-11-06 15:07:21', NULL, 0);
```
Comment (mbayopanda) [inline test/data.sql:244]: 👍 great

File: test/data.sql [hunk 9]
```diff
@@ -36,245 +254,290 @@ INSERT INTO user (id, username, password, display_name, email, deactivated) VALU
...
-(139, 1),
-
--- inventory configuration module
-(140, 1),
-
--- Voucher records
-(141, 1),
+-- the super user has permission to everything user
+INSERT INTO permission (unit_id, user_id)
+SELECT unit.id, @superUser FROM unit ON DUPLICATE KEY UPDATE unit_id = unit_id, user_id = user_id;
```
Comment (mbayopanda) [inline test/data.sql:261]: 👍 very nice

File: test/data.sql [hunk 10]
```diff
@@ -36,245 +254,290 @@ INSERT INTO user (id, username, password, display_name, email, deactivated) VALU
...
+
+SET @USD = 1;
+SET @FC = 2;
 
 -- exchange rate for the current date
 INSERT INTO `exchange_rate` VALUES
-  (1,1,1,900.0000, DATE('2016-01-01')),
-  (2,1,1,930.0000, NOW());
+  (1, 1, @USD, 900.0000, DATE('2016-01-01')),
+  (2, 1, @USD, 930.0000, NOW());
```
Comment (mbayopanda) [inline test/data.sql:296]: 👍

File: test/data.sql [hunk 11]
```diff
@@ -36,245 +254,290 @@ INSERT INTO user (id, username, password, display_name, email, deactivated) VALU
...
+  (1, HUID('0761a3c9-8c39-4741-9629-b2b55ea832a6'), '160001',   'Amethocaine 0.5%Flc 5cc',                                0.8100,   1, HUID('8d85a207-7c04-4d60-8891-fec7e5774b3d'), 2,  0, 0, 1, 100000000, 0, 1, 1, 0, 1, 1.0000, 1.0000, '2016-10-23 08:35:08', NULL),
+  (1, HUID('a0525113-8177-4e08-bc38-9147af93b9be'), '160002',   'Atropine 1%Flc 5cc',                                     0.9400,   1, HUID('8d85a207-7c04-4d60-8891-fec7e5774b3d'), 2,  0, 0, 1, 100000000, 0, 1, 1, 0, 1, 1.0000, 1.0000, '2016-10-23 08:35:08', NULL),
+  (1, HUID('07dc0762-c9a6-4e67-a08f-ccfae34f7f7e'), '170448',   'Nystatin 100.000UI/ml suspension orale 30ml fl',         1.5000,   1, HUID('4f804446-16cc-47e4-a40f-8d915107d431'), 10, 0, 0, 0, 0,         0, 1, 1, 0, 1, 1.0000, 1.0000, '2016-10-23 08:35:08', NULL),
+  (1, HUID('6fd1e156-e703-4c95-b64d-65859464fd99'), '170449',   'Nevirapine (NVP) Sirop 10mg/ml 240ml fl',                5.4534,   1, HUID('4f804446-16cc-47e4-a40f-8d915107d431'), 10, 0, 0, 0, 0,         0, 1, 1, 0, 1, 1.0000, 1.0000, '2016-10-23 08:35:08', NULL);
```
Comment (mbayopanda) [inline test/data.sql:519]: 💃 great

**Review by mbayopanda — APPROVED** [review]
This PR is very useful, it is a great improvement for the  data.sql to use mysql variables.

Good job

File: client/src/modules/inventory/configuration/groups/groups.js (hunk 3 above)
Comment (jniles) [reply client/src/modules/inventory/configuration/groups/groups.js:56]: The modal is sometimes rejected with no error message (such as when `ModalInstance.dismiss()` is called).  So, we must check if the error has occurred or if it is just a dismissed modal instance.

**Pull request merged**